plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/helpers.ts

```diff
@@ -313,6 +313,115 @@ export async function initRepoPushAndProtect(
   }
 }
 
+export async function createGithubTemplateRepoWithCollaboratorsAndTopics(
+  client: Octokit,
+  repo: string,
+  owner: string,
+  repoVisibility: 'private' | 'public',
+  description: string | undefined,
+  templateOwner: string,
+  templateRepo: string,
+  access: string | undefined,
+  collaborators:
+    | (
+        | {
+            user: string;
+            access: 'pull' | 'push' | 'admin' | 'maintain' | 'triage';
+          }
+        | {
+            team: string;
+            access: 'pull' | 'push' | 'admin' | 'maintain' | 'triage';
+          }
+        | {
+            /** @deprecated This field is deprecated in favor of team */
+            username: string;
+            access: 'pull' | 'push' | 'admin' | 'maintain' | 'triage';
+          }
+      )[]
+    | undefined,
+  logger: Logger,
+) {
+  // eslint-disable-next-line testing-library/no-await-sync-query
+  const user = await client.rest.users.getByUsername({
+    username: owner,
+  });
+
+  const templateRepoCreationPromise = client.rest.repos.createUsingTemplate({
+    name: repo,
+    owner: owner,
+    private: repoVisibility === 'private',
```

Comment:
The problem with this REST API is that it does not support creating a repository based on a template and setting the visibility to ``internal``. This visibility level seems to only be available via the GraphQL API. 

You can solve it with a function something like this:
```TypeScript
/**
 * Creates a repository based on a GitHub Template
 *
 * @param client - An octokit graphql client
 * @param logger - A Winston Logger from the action context
 * @param repositoryId - Id of the template repository
 * @param name - Name of the new repository
 * @param visibility - Visibility of the new repository - Internal, Public or Private
 * @param ownerId - Id of the owner
 * @param description - Description of the repository. Optional
 */
export async function createRepoBasedOnTemplate(
  client: any,
  logger: Logger,
  repositoryId: string,
  name: string,
  visibility: string,
  ownerId: string,
  description?: string,
): Promise<Repository> {
  const mutation = `
    mutation CreateRepoBasedOnTemplate($repositoryId: ID!, $name: String!, $visibility: RepositoryVisibility!, $ownerId: ID!, $description: String) {
        cloneTemplateRepository(input: {clientMutationId:"backstage-app", repositoryId: $repositoryId, name: $name, visibility: $visibility, ownerId: $ownerId, description: $description}) {
          repository {
            id
            name
            nameWithOwner
            url
          }
        }
      }`;

  let repository: Repository;
  try {
    const createRepoPromise = await client.graphql(mutation, {
      repositoryId,
      name,
      visibility,
      ownerId,
      description,
    });
    repository = (await createRepoPromise).cloneTemplateRepository.repository;
  } catch (error) {
    assertError(error);
    if (error.message === 'Resource not accessible by integration') {
      logger.warn(
        `The GitHub app or token provided may not have the required permissions to create the repository.`,
      );
    }
    throw new Error(`Failed to create the repository, ${error.message}`);
  }
  return repository;
}
```